python/dgl/data/movielens.py

```diff
@@ -0,0 +1,640 @@
+                shuffle=True,
+            )
+            train_rating_data, valid_rating_data = (
+                train_rating_data.iloc[train.indices],
+                train_rating_data.iloc[valid.indices],
+            )
+            all_rating_data = pd.concat(
+                [train_rating_data, valid_rating_data, test_rating_data]
+            )
+
+        elif self.name == "ml-1m" or self.name == "ml-10m":
+            all_rating_data = self._load_raw_rates(
+                os.path.join(self.raw_path, "ratings.dat"), "::"
+            )
+            indices = np.arange(len(all_rating_data))
+            train, valid, test = split_dataset(
+                indices,
+                [
+                    1 - self.valid_ratio - self.test_ratio,
+                    self.valid_ratio,
+                    self.test_ratio,
+                ],
+                shuffle=True,
+            )
+            train_rating_data, valid_rating_data, test_rating_data = (
+                all_rating_data.iloc[train.indices],
+                all_rating_data.iloc[valid.indices],
+                all_rating_data.iloc[test.indices],
+            )
+
+        # 2. load user and movie data, and drop those unseen in rating_data
+        user_data = self._load_raw_user_data()
+        movie_data = self._load_raw_movie_data()
+        user_data = self._drop_unseen_nodes(
+            data_df=user_data,
+            col_name="id",
+            reserved_ids_set=set(all_rating_data["user_id"].values),
+        )
+        movie_data = self._drop_unseen_nodes(
+            data_df=movie_data,
+            col_name="id",
+            reserved_ids_set=set(all_rating_data["movie_id"].values),
+        )
+
+        self.user_feat = tensor(self._process_user_fea(user_data)) 
+
+        # 3. generate rating pairs
+        # Map user/movie to the global id
+        self._global_user_id_map = {
+            ele: i for i, ele in enumerate(user_data["id"])
+        }
+        self._global_movie_id_map = {
+            ele: i for i, ele in enumerate(movie_data["id"])
+        }
+
+        # pair value is idx rather than id
+        (
+            train_u_indices,
+            train_v_indices,
+            train_labels,
+        ) = self._generate_pair_value(train_rating_data)
+        (
+            valid_u_indices,
+            valid_v_indices,
+            valid_labels,
+        ) = self._generate_pair_value(valid_rating_data)
+        test_u_indices, test_v_indices, test_labels = self._generate_pair_value(
+            test_rating_data
+        )
+
+        # reindex u and v, v nodes start after u
+        num_user = len(self._global_user_id_map)
+        train_v_indices += num_user
+        valid_v_indices += num_user
+        test_v_indices += num_user
+
+        self.train_rating_pairs = (
+            tensor(train_u_indices, dtype=data_type_dict["int64"]),
+            tensor(train_v_indices, dtype=data_type_dict["int64"]),
+        )
```

Comment:
You can simply use `torch.LongTensor` since you are using PyTorch anyway. Same for all other usage of framework-agnostic APIs.

Reply:
Ok. fix